nio/rooms.py

```diff
@@ -97,29 +100,44 @@ def group_name(self):
         """
         # Sort user display names, excluding our own user and using the
         # mxid as the sorting key.
-        #
-        # TODO: Hook the user display name disambiguation algorithm here.
-        # Currently, we use the user display names as is, which may not be
-        # unique.
-        users = [
-            user.user_id
-            for mxid, user in sorted(self.users.items(), key=lambda t: t[0])
-            if mxid != self.own_user_id
-        ]
-
-        num_users = len(users)
+
+        user_names = [self.user_name(u) for u in sorted(self.users.keys())]
```

Comment:
The comment states "excluding our own user", but you removed the if statement filtering out our own user.

The group name shouldn't display our own name as far as i can remember.
Looks good otherwise. Just add the if back and I'll merge this.

Reply:
fixed and added a test